### PR TITLE
[Horizon] Add completion checkmark

### PR DESCRIPTION
**Pull request by Ahmed-Naguib93**
Add module completion checkmark
https://instructure.atlassian.net/browse/CLX-149

### CODE REVIEW

**Comment by inst-danger** [comment]
<details><summary>Horizon Build QR Code:</summary>

</details>

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>One or more files are below the minimum test coverage 50%</td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>The total test coverage is below the minimum 90%</td>
    </tr>
  </tbody>
</table>


Coverage | New % | Master % | Delta
-------- | ----- | -------- | -----
**Canvas iOS** | 89.99% | 90.05% | -0.06%
Core/Core/Courses/CourseProgression/CDCourseProgression.swift | 0% | -- | --
Core/Core/Planner/CalendarMain/ViewModel/PlannerViewModel.swift | 0% | 0% | 0%
Core/Core/Planner/CalendarMain/Model/CalendarWeek.swift | 0% | 0% | 0%
Core/Core/Planner/CalendarMain/View/Calendar/CalendarCardInteractionState.swift | 0% | 0% | 0%
Core/Core/Courses/CourseProgression/GetCoursesProgressionRequest.swift | 0% | -- | --
Core/Core/Planner/CalendarMain/Model/PlannablesInteractor.swift | 0% | 0% | 0%
Core/Core/Planner/CalendarMain/View/Calendar/ViewPreferences.swift | 0% | 0% | 0%
Core/Core/Planner/CalendarMain/Model/CalendarDay.swift | 0% | 0% | 0%
Core/Core/Planner/CalendarMain/Model/CalendarMonth.swift | 0% | 0% | 0%
Core/Core/Planner/CalendarMain/View/Calendar/CalendarUtils.swift | 0% | 0% | 0%
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against b5f8dc73cb9c2c72de086ebe6a78b0c82bfda54a
</p>

**Review by szabinst — CHANGES_REQUESTED** [review]
LGTM functionality wise!

File: Horizon/Horizon/Sources/Common/View/ExpandingModuleView.swift [hunk 1]
```diff
@@ -42,40 +52,87 @@ struct ExpandingModuleView: View {
...
+            Circle()
+                .fill(Color.backgroundLightest)
+                .frame(width: 25, height: 25)
+            ProgramCheckMarkIcon(isCompleted: isCompleted)
+
+        }
     }
 }
 
+#if DEBUG
```
[the remark this answers is not in the document]
Comment (Ahmed-Naguib93) [reply Horizon/Horizon/Sources/Common/View/ExpandingModuleView.swift:126]: Removed

File: Horizon/Horizon/Sources/Common/View/ProgramItemOverdueView.swift [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+            Text("OVERDUE", bundle: .horizon)
+                .font(.regular12)
+                .foregroundColor(Color.textDanger)
+                .padding(5)
+                .overlay(Capsule().stroke(Color.backgroundDanger, lineWidth: 1))
+        }
+    }
+}
+
+#if DEBUG
```
[the remark this answers is not in the document]
Comment (Ahmed-Naguib93) [reply Horizon/Horizon/Sources/Common/View/ProgramItemOverdueView.swift:48]: Removed

**Review by szabinst — APPROVED** [review]

**Pull request merged**